modules/apps/batch-planner/batch-planner-test/src/testFunctional/tests/ImportAndExportObjects.testcase

```diff
@@ -0,0 +1,267 @@
+				deletionType = "cascade",
+				name = "universityStudents",
+				parentObjectName = "University",
+				type = "oneToMany");
+		}
+
+		task ("And Given one entry related to antoher one") {
+			var universityEntryId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "universities",
+				name = "Oxford");
+
+			ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "students",
+				name = "Pearl");
+
+			ObjectDefinitionAPI.relateObjectEntriesByName(
+				child_plural_label = "students",
+				childEntryName = "Pearl",
+				parent_plural_label = "universities",
+				parentEntryName = "Oxford",
+				relationshipName = "universityStudents");
+		}
+
+		task ("When user export custom object as entity type") {
+			ImportExport.openImportExportAdmin();
+
+			ImportExport.exportFile(
+				entityType = "C_Student (v1_0 - Liferay Object REST)",
+				exportFields = "Attribute Code",
+				exportFileFormat = "JSON");
+		}
+
+		task ("Then zip file contains related file type export successfully") {
+			ImportExport.unzipDownloadedExportFile();
+
+			BatchEngine.assertExportedFileContainsCorrectObject(
+				expectedValue = "Pearl",
+				fileName = "export.json",
+				jsonObject = "$..[0]name");
+		}
+	}
+
+	@priority = 4
+	test ExportCustomObjectWithAllFieldTypesMapped {
+		property portal.acceptance = "true";
+
+		task ("Given a custom object with all kinds of field type is publushed") {
+			ObjectDefinitionAPI.createObjectDefinitionWithAllKindsOfFieldTypes(
+				en_US_label = "student",
+				en_US_plural_label = "students",
+				objectDefinitionName = "Student",
+				requiredStringFieldName = "name");
+		}
+
+		task ("And Given object entry is created") {
+			var studentEntryId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "students",
+				name = "Jessica");
+
+			ObjectFieldAPI.updateAllFieldsOfEntryById(
+				customBooleanValue = "true",
+				customLongTextValue = "This is a custom LongText field",
+				customPicklistValue = "distance1",
+				customPrecisionDecimalValue = "12.55",
+				en_US_plural_label = "students",
+				entryId = ${studentEntryId},
+				requiredFieldLabel = "name",
+				requiredFieldValue = "Jessica");
+
+			ObjectAdmin.selectAttachmentFileInObjectEntryViaUI(
+				fileName = "Document_3.png",
+				key_entry = ${studentEntryId},
+				label = "student",
+				objectName = "Student");
+		}
+
+		task ("When user export custom object with all attributes codes selected") {
+			ImportExport.openImportExportAdmin();
+
+			ImportExport.exportFile(
+				entityType = "C_Student (v1_0 - Liferay Object REST)",
+				exportFields = "name,creator,customAttachment,customBoolean,customLongText,customPicklist,customPrecisionDecimal,",
+				exportFileFormat = "JSON");
+		}
+
+		task ("Then the zip file contains a related file type export successfully") {
+			ImportExport.assertFileContent(
+				comparisonFileName = "json_objectDefinitionEntry_export.txt",
```

Comment:
What about rename json_objectDefinitionEntry_export.txt file? My concerns:
- json_ shoud ends with json also 
- objectDefinitionEntry to objectEntry

Reply:
I don't think so, we need to use json_name.txt, if we change to .json the method assertFileContent will fail, because the export file content format doesn't as same as the JSON file